Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs

```diff
@@ -27,14 +27,28 @@ public string Accentuate(string message, FrenchAccentComponent component)
 
         msg = _replacement.ApplyReplacements(msg, "french");
 
-        // replaces th with z
-        msg = RegexTh.Replace(msg, "'z");
-
         // replaces h with ' at the start of words.
         msg = RegexStartH.Replace(msg, "'");
 
         // spaces out ! ? : and ;.
         msg = RegexSpacePunctuation.Replace(msg, " $&");
+        
+        // replaces th with 'z or 's depending on the case
+        foreach (Match match in RegexTh.Matches(msg))
+        {
+            var uppercase = msg.Substring(match.Index, 2).Contains("TH");
+            var Z = uppercase ? "Z" : "z";
+            var S = uppercase ? "S" : "s";
+            var idxLetter = match.Index + 2;
+
+            // If th is alone, just do 'zis
```

Comment:
```suggestion
            // If th is alone, just do 'z
```
is what this is doing, right? The extra `is` in the comment threw me off.

Reply:
maybe I shouldn't put jokes in my comments. Yes, that's what it does